src/model/create-model.ts

```diff
@@ -96,7 +96,7 @@ export const _buildModel = (metadata: ModelMetadata) => {
       const strategy = options.strategy || CAST_STRATEGY.DEFAULT_OR_DROP;
       const strict = options.strict !== undefined ? options.strict : schema.options.strict;
       const skip = options.skip || [modelKey, ID_KEY];
-      const schemaData = cast(data, schema, { strategy, strict, skip });
+      const schemaData = cast(data, schema, { strategy, strict: !!strict, skip });
```

Comment:
You have the wrong concept of strict. Review the documentation in this section https://v2.ottomanjs.com/guides/schema.html#cast-method